src/toMatchOneOf.ts

```diff
@@ -52,7 +52,7 @@ export function toMatchOneOf<T extends {}>(
 ): JestResult {
   if (isArray(received)) {
     if (received.length === 0) {
-      if (some(expectedValues, isArray)) {
+      if (some(expectedValues, isArray) || expectedValues.length === 0) {
```

Comment:
Im not sure this is quite right, 
E.g this will cause the following to succeed:

```
expect({ one: 1, two: 2}).toMatchOneOf([])
```
Which I think makes sense to be a failure (the test author probably forgot to fill out the expected values).

I think If you add a test case that more closely matches the one you reported in your issue this will become apparent:

```
it('will accept values when the expected is an empty array', () => {
  expect({ a: []}).toMatchOneOf([{ a: []}])
})
```

Reply:
I tested the test case, and it doesn't pass
```
    expect(received).toMatchShapeOf(expected)

    For:
      type: object
      value: {"one": 1, "two": 2}
    Expected type to be one of

      53 | 
      54 |   it('testing', () => {
    > 55 |     expect({ one: 1, two: 2}).toMatchOneOf([])
         |                               ^
      56 |   })
```

Reply:
And for some reason, I expect type is Array<any>.
```
  it('passes when expect Array<any> in test', () => {
    interface Test {
      test: Array<any>
    }

    const checkValue: Test = {
      test: [2, 3, 4]
    }

    const expectedValue: Test = {
      test: []
    }

    expect(checkValue).toMatchShapeOf(expectedValue)
  })
```
but it is conflict to the existing test case:
```
  it('will not accept values when the expected is an empty array', () => {
    expect({ a: [1, 2, 3]}).not.toMatchOneOf([{ a: []}])
  })
```
I think it should allow when expected type is Array<any>
```
    if (expectedValues.length === 0) {
      return successResult
    }
```

Reply:
Looks like I was a little mistaken in my initial reply, and your original addition was exactly what was needed (sorry!).

I disagree with the additions in the latest commit though, I think the following
```
    const received: Test = { test: [2, 3, 4] }
    const expected: Test = { test: [] }
    expect(checkValue).toMatchShapeOf(expected)
```
Should *fail* because:
- The test author probably made a mistake filling out `expected` (maybe they forgot, or managed to get `received` and `expected` the wrong way round.
- If you don't want to check the values in `test`, you have to option of doing 
```
    const received: Test = { a: 1, test: [2, 3, 4] }
    const expected: Test = { a: 1 }
    expect(checkValue).toMatchShapeOf(expected)
```
which will pass.

I have created a PR into your fork: https://github.com/skmdev/jest-to-match-shape-of/pull/1
So if you agree you can accept those changes and I will merge this PR.

If not can you provide an example where the above behaviour isn't what you would want?

Thanks again for your efforts.

Reply:
I agree with you, but I have a question on it.

For example:
If the endpoint will response a list of photo
```
interface Response {
  photos: Array<{
     url: string;
  }>
}

// contains photo
const havePhoto: Response = {
   photos: [
     { url: "test" }
   ]
}

// empty photo
const empty: Response = {
   photos: []
}
```
The response will probably return an empty array, and I cannot ensure the `photos` contains photo or not.

Test case:
```
    const received: Test = { photos: [] }
    const expected: Test = { photos: [{ url: 'test' }] }
    expect(checkValue).toMatchShapeOf(expected)
```

When `received.photos` is an empty array, it will occurs error if I put the shape in array.

Since there is no optional type in array at this moment, I skip to check photos type in this moment

Reply:
I think I would call this a bug, and I would expect your last test case to pass.
I should have some time to look at this tonight.

Reply:
Thanks a lot.